### ISSUE TITLE
AudioServerSW::update() : is this a bug ?

### ISSUE THREAD

**Issue by SuperUserNameMan** [godotengine/godot#7264]
**Issue description** (what happened, and what was expected):

I was giving a look at the AudioServer guts, and i've found something bizarre here :
https://github.com/godotengine/godot/blob/master/servers/audio/audio_server_sw.cpp#L848

````
void AudioServerSW::update() {

	_update_streams(false);
#ifdef NO_THREADS

	_update_streams(true);
#endif
}
````

According to the constructor of `void AudioServerSW::_update_streams(bool p_thread)`, shouldn't it be something like this instead ?

````
void AudioServerSW::update() {
#ifdef NO_THREADS
	_update_streams(false);
#else
	_update_streams(true);
#endif
}
````

I recompiled and made a tiny test which seems to make no difference

**Comment by karroffel** [comment]
Shouldn't it be
```cpp
#ifdef NO_THREADS
        _update_streams(true);
#else
        _update_streams(false);
#endif
```
?

Also, is this still relevant with the new audio engine in 3.0 or is that relevant for 2.x as well?

**Comment by reduz** [comment]
audio engine in 3.0 is completely rewritten

**Comment by bojidar-bg** [comment]
Is this still an issue?

**Comment by Noshyaar** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
Not relevant in 3.0+.
